Sources/SwiftCliCore/ChartConstructor.swift

```diff
@@ -38,71 +38,94 @@ public class ChartConstructor {
         return(finishedChart)
     }
     public func makeTopRow(width: Int) -> String {
-        var topRowBars = ""
-        if width > 0 {
-            topRowBars += "┌"
-            if width > 1 {
-                topRowBars += String(repeating: "─┬", count: width - 1)
-
-            }
-            topRowBars += "─┐"
+        switch width {
+        case 0:
+            return "\n"
+        case 1:
+            return "┌─┐\n"
+        default:
+            let middleBoxes = String(repeating: "─┬", count: width - 1)
+            return "┌\(middleBoxes)─┐\n"
         }
-        topRowBars += "\n"
-        return(topRowBars)
+
     }
     public func makeMiddleRow(width: Int) -> String {
-        var middleRowBars = ""
-        if width > 0 {
-            middleRowBars += "├"
-            if width > 1 {
-                middleRowBars += String(repeating: "─┼", count: width - 1)
-            }
-            middleRowBars += "─┤"
+        switch width {
+        case 0:
+            return "\n"
+        case 1:
+            return "├─┤\n"
+        default:
+            let middleBoxes = String(repeating: "─┼", count: width - 1)
+            return "├\(middleBoxes)─┤\n"
         }
-        middleRowBars += "\n"
-        return(middleRowBars)
     }
 
     public func makeMiddleRowStitchCountChange(width: Int, rDiff: Int, lDiff: Int) -> String {
+        var leftStitches = ""
+        var middleStitches = ""
+        var rightStitches = ""
 
-        var middleRowBars = ""
-        if width > 0 {
-            middleRowBars += "├"
-            if rDiff < 0 {
-                middleRowBars += String(repeating: "─┼", count: (width))
-                if abs(rDiff) >= 2 {
-                    middleRowBars += String(repeating: "─┬", count: ((abs(rDiff))-1))
 
-                }
-                middleRowBars += "─┐"
-            }
+        switch lDiff {
+        case 0:
+            leftStitches = "├"
+        default:
+            leftStitches = "├"
+        }
+
+        switch width {
+        case 1:
+            middleStitches = "─"
+        default:
+            middleStitches = String(repeating: "─┼", count: (width-1))
+        }
+
+
+        switch (true){
+        case (rDiff >= 1):
+            rightStitches = ""
+        case (rDiff == -1):
+            rightStitches = "─┼─┐\n"
+        case (rDiff < -1):
+            let midRightSt = String(repeating: "─┬", count: ((abs(rDiff))-1))
+            rightStitches = "─┼\(midRightSt)─┐\n"
+        default:
+            rightStitches = "─┤\n"
```

Comment:
I wonder if it'll make more sense to order these cases as `rDiff < -1` first, then `rDiff == -1`, then `rDiff == 0`, then make the `rDiff >= 1` be default rather than the separate case.